### PR TITLE
backend/feat/#1672 get-ticket-api

### PR DESCRIPTION
**Pull request by JanhaviSoni**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Ride.hs [hunk 1]
```diff
@@ -392,22 +392,23 @@ findAllRideItems merchantID limitVal offsetVal mbBookingStatus mbRideShortId mbC
...
-                    B.&&?. (maybe (B.sqlBool_ $ B.val_ True) (\defaultTo -> B.sqlBool_ $ ride.createdAt B.<=. B.val_ (defaultTo)) mbTo)
-                    B.&&?. (maybe (B.sqlBool_ $ B.val_ True) (\bookingStatus -> mkBookingStatusVal ride B.==?. B.val_ (bookingStatus)) mbBookingStatus)
-              )
-              -- B.&&?. B.ifThenElse_ (ride.status B.==. B.val_ Ride.NEW) (booking.status B.val_ Common.RCOMPLETED) (B.val_ Common.RCANCELLED)) $
-              do
-                booking' <- B.all_ (BeamCommon.booking BeamCommon.atlasDB)
-                ride' <- B.join_' (BeamCommon.ride BeamCommon.atlasDB) (\ride'' -> BeamR.bookingId ride'' B.==?. BeamB.id booking')
-                person' <- B.join_' (BeamCommon.person BeamCommon.atlasDB) (\person'' -> BeamP.id person'' B.==?. BeamB.riderId booking')
-                pure (booking', ride', person')
+            B.orderBy_ (\(_, ride, _) -> B.desc_ ride.createdAt) $
```
Comment (hkmangla) [inline Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Ride.hs:395]: Why did you change this query?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs [hunk 2]
```diff
@@ -112,7 +115,47 @@ buildRideListItem QRide.RideItem {..} = do
...
+      Just number -> do
+        no <- getDbHash number
+        let code = fromMaybe "+91" countryCode
+        person <- runInReplica $ QPerson.findByMobileNumberAndMerchant code no merchant.id >>= fromMaybeM (PersonWithPhoneNotFound number)
+        ridesAndBooking <- QRide.findAllByDriverId person.id (Just totalRides) (Just 0) Nothing Nothing Nothing
+        let lastNRides = map fst ridesAndBooking
+        ridesDetail <- mapM (\ride -> rideInfo merchantShortId (cast ride.id)) lastNRides
+        let rdList = zipWith (curry (makeRequiredRideDetail person.id)) lastNRides ridesDetail
+        return Common.TicketRideListRes {rides = rdList}
+      Nothing -> throwError $ InvalidRequest "Invalid Request"
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs:139]: Throw proper error. You should always throw proper error.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs [hunk 3]
```diff
@@ -112,7 +115,47 @@ buildRideListItem QRide.RideItem {..} = do
       }
 
 ---------------------------------------------------------------------------------------------------
+ticketRideList :: ShortId DM.Merchant -> Maybe (ShortId Common.Ride) -> Maybe Text -> Maybe Text -> Maybe Text -> Flow Common.TicketRideListRes
+ticketRideList merchantShortId mbRideShortId countryCode mbPhoneNumber _ = do
+  merchant <- findMerchantByShortId merchantShortId
+  let totalRides = 5
+  let mbShortId = coerce @(ShortId Common.Ride) @(ShortId DRide.Ride) <$> mbRideShortId
+  case mbShortId of
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs:123]: Do like this `(mbShortId, mbPhoneNumber)`

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Ride.hs (hunk 1 above)
Comment (JanhaviSoni) [reply Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Ride.hs:395]: It is because I wanted the same result in descending order so in place of fetching and sorting outside or making any new query it would be better here only in this query

Comment (hkmangla) [reply Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Ride.hs:417]: Sorting will make this query very slow. Lets remove sorting

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by hkmangla — APPROVED** [review]
